flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaConnectorOptionsUtil.java

```diff
@@ -535,6 +538,15 @@ private static void autoCompleteSubject(
         }
     }
 
+    static void validateDeliveryGuarantee(ReadableConfig tableOptions) {
+        if (tableOptions.getOptional(DELIVERY_GUARANTEE).orElse(DELIVERY_GUARANTEE.defaultValue())
+                        == DeliveryGuarantee.EXACTLY_ONCE
+                && !tableOptions.getOptional(TRANSACTIONAL_ID_PREFIX).isPresent()) {
+            throw new ValidationException(
+                    "sink.transactional-id-prefix must be specified when using DeliveryGuarantee.EXACTLY_ONCE.");
```

Comment:
nit: use `DELIVERY_GUARANTEE.key()`

Reply:
Is there any way to generate a kinda unique string from SQL that survives manual restarts? @twalthr 
Then we could also convert the error into a warning and assign this transaction id.

Reply:
I'm sure we could create one with a lot of parameters such as configuration and plan, but manually specifying it might be the best option for now. Would it make sense to allow this on a per record basis using a metadata column, I guess not? And job graph id is not an option?

Reply:
I guess we can leave that as a future improvement. We could then also use it for datastream.